lib/transforms/subsetFonts.js

```diff
@@ -921,54 +919,65 @@ These glyphs are used on your site, but they don't exist in the font you applied
-        cssAssetMap[subsetCssText] = cssAsset;
+      let cssAsset = assetGraph.addAsset({
+        type: 'Css',
+        url: `${subsetUrl}subfontTemp.css`,
+        text: subsetCssText
+      });
 
-        if (!inlineSubsets) {
-          for (const fontRelation of cssAsset.outgoingRelations) {
-            const fontAsset = fontRelation.to;
-            const extension = fontAsset.contentType.split('/').pop();
-
-            const nameProps = ['font-family', 'font-weight', 'font-style']
-              .map(prop =>
-                fontRelation.node.nodes.find(decl => decl.prop === prop)
-              )
-              .map(decl => decl.value);
-
-            const fileNamePreFix = `${unquote(nameProps[0])
-              .replace(/__subset$/, '')
-              .replace(/ /g, '_')}-${normalizeFontPropertyValue(
-              'font-weight',
-              nameProps[1]
-            )}${nameProps[2] === 'italic' ? 'i' : ''}`;
-
-            const fontFileName = `${fileNamePreFix}-${fontAsset.md5Hex.slice(
-              0,
-              10
-            )}.${extension}`;
-
-            fontAsset.url = subsetUrl + fontFileName;
-
-            if (inlineCss) {
-              fontRelation.hrefType = 'rootRelative';
-            } else {
-              fontRelation.hrefType = 'relative';
-            }
+      if (!inlineSubsets) {
+        for (const fontRelation of cssAsset.outgoingRelations) {
+          const fontAsset = fontRelation.to;
+          const extension = fontAsset.contentType.split('/').pop();
+
+          const nameProps = ['font-family', 'font-weight', 'font-style']
+            .map(prop =>
+              fontRelation.node.nodes.find(decl => decl.prop === prop)
+            )
+            .map(decl => decl.value);
+
+          const fileNamePreFix = `${unquote(nameProps[0])
+            .replace(/__subset$/, '')
+            .replace(/ /g, '_')}-${normalizeFontPropertyValue(
+            'font-weight',
+            nameProps[1]
+          )}${nameProps[2] === 'italic' ? 'i' : ''}`;
+
+          const fontFileName = `${fileNamePreFix}-${fontAsset.md5Hex.slice(
+            0,
+            10
+          )}.${extension}`;
+          const fontUrl = subsetUrl + fontFileName;
+
+          const [identicalAsset] = assetGraph.findAssets({ url: fontUrl });
+
+          if (identicalAsset) {
+            fontRelation.to = identicalAsset;
+          } else {
+            fontAsset.url = fontUrl;
+          }
+
+          if (inlineCss) {
+            fontRelation.hrefType = 'rootRelative';
+          } else {
+            fontRelation.hrefType = 'relative';
           }
         }
+      }
+
+      const cssFileName = `fonts-${cssAsset.md5Hex.slice(0, 10)}.css`;
+      const cssUrl = subsetUrl + cssFileName;
+      const [identicalAsset] = assetGraph.findAssets({ url: cssUrl });
 
-        const cssFileName = `fonts-${cssAsset.md5Hex.slice(0, 10)}.css`;
-        cssAsset.url = subsetUrl + cssFileName;
+      if (identicalAsset) {
```

Comment:
Would it be possible to detect this condition earlier so we could avoid creating and adding the duplicate to the graph?

Reply:
I think the answer is yes. But the current strategy would need a big revision. The part that gets the relevant stylesheet currently gets a string with base 64 inlined fonts. There was an attempt to dedupe on the per-page stylesheet level, but that failed and also seems like the wrong place to ensure this

Reply:
I also need a test to verify. I thought a current one already covered this, but alas. Need to figure out how to trigger this condition

Reply:
Okay, cool. As long as a test is included, it will help us not reintroduce the bug when we get around to doing that revision :)